### PR TITLE
Add test cases when attributes are null to "survey update" API

### PR DESCRIPTION
**Pull request by DarrenZhang01**
Add tests to include the cases where some of the attributes are null or all the attributes are null in the data package sent to the "survey update" post API

### CODE REVIEW

File: test/back_end/api_post_test.spec.js [hunk 1]
```diff
@@ -86,6 +86,88 @@ describe("Test POST/UPDATE/DELETE requests to API", function() {
...
 
+        it("send data package with no attributes to the 'update survey setting' API", async function () {
+            try {
+                let fetched = await fetch(
+                    "http://localhost:3000/post_info.php?what=surveys&survey_id=1&user_id=butler84&level=section&action=add_or_update"
+                );
+                expect(fetched).to.have.status(200);
+                let fetchedJSON = await fetched.json();
+                expect(fetchedJSON.TYPE).to.be.equal("success");
+                expect(fetchedJSON.DATA).to.be.equal(null);
```
Comment (siefkenj) [inline test/back_end/api_post_test.spec.js]: Don't include these two lines. We're going to be changing the API

File: test/back_end/api_post_test.spec.js [hunk 2]
```diff
@@ -86,6 +86,88 @@ describe("Test POST/UPDATE/DELETE requests to API", function() {
...
+
+                // Check if the "timedate_open" and "timedate_close" are changed in the database to what we revised
+                fetched = await fetch(
+                    "http://localhost:3000/get_info.php?what=surveys&survey_id=1&user_id=woods13"
+                );
+                expect(fetched).to.have.status(200);
+                fetchedJSON = await fetched.json();
+                expect(fetchedJSON.TYPE).to.be.equal("survey_package");
+                expect(fetchedJSON.DATA[0].timedate_open).to.be.equal(timedate_open);
+                expect(fetchedJSON.DATA[0].timedate_close).to.be.equal(timedate_close);
```
Comment (siefkenj) [inline test/back_end/api_post_test.spec.js]: Good!

File: test/back_end/api_post_test.spec.js [hunk 3]
```diff
@@ -86,6 +86,88 @@ describe("Test POST/UPDATE/DELETE requests to API", function() {
...
+                let fetchedJSON = await fetched.json();
+                expect(fetchedJSON.TYPE).to.be.equal("success");
+                expect(fetchedJSON.DATA).to.be.equal(null);
+            } catch (error) {
+                throw error;
+            }
+        });
+
+        it("send data package with only time setting attributes to the 'update survey setting' API", async function () {
+            try {
```
Comment (siefkenj) [inline test/back_end/api_post_test.spec.js]: Don't put this in a `try..catch` block

File: test/back_end/api_post_test.spec.js [hunk 4]
```diff
@@ -86,6 +86,88 @@ describe("Test POST/UPDATE/DELETE requests to API", function() {
...
+                    default_survey_close: timedate_close
+                });
+                let post_body = base64.encode(body);
+                let fetched = await fetch(
+                    "http://localhost:3000/post_info.php?what=surveys&survey_id=1&user_id=butler84&level=section&action=add_or_update&post_body=base64:" + post_body
+                );
+                expect(fetched).to.have.status(200);
+                let fetchedJSON = await fetched.json();
+                expect(fetchedJSON.TYPE).to.be.equal("success");
+                expect(fetchedJSON.DATA).to.be.equal(null);
```
Comment (siefkenj) [inline test/back_end/api_post_test.spec.js]: Don't verify these things. The API will change.
